src/common/utils/setup-helpers.js

```diff
@@ -328,18 +328,41 @@ function checkNativeTextTracksSupport(options: KPOptionsObject): void {
   }
 }
 
+/**
+ * Sets config option for Ads with MSE
+ * @private
+ * @param {KPOptionsObject} options - kaltura player options
+ * @returns {void}
+ */
+function _configureAdsWithMSE(options: KPOptionsObject): void {
+  const playAdsWithMSE = Utils.Object.getPropertyPath(options, 'playback.playAdsWithMSE');
+  //dai should play without playAdsWithMSE config
+  if (typeof playAdsWithMSE !== 'boolean') {
+    if (options.plugins && options.plugins.imadai && !options.plugins.imadai.disable) {
+      options = Utils.Object.createPropertyPath(options, 'playback.playAdsWithMSE', false);
+    } else {
+      options = Utils.Object.createPropertyPath(options, 'playback.playAdsWithMSE', true);
+    }
+  }
+  const disableMediaPreloadIma = Utils.Object.getPropertyPath(options, 'plugins.ima.disableMediaPreload');
+  const disableMediaPreloadBumper = Utils.Object.getPropertyPath(options, 'plugins.bumper.disableMediaPreload');
+
+  if (options.plugins && options.plugins.ima && typeof disableMediaPreloadIma !== 'boolean') {
```

Comment:
```suggestion
  if (typeof disableMediaPreloadIma !== 'boolean') {
```
`createPropertyPath` already validates the path , same for bumper.

Reply:
I don't want to validate it...
If IMA doesn't exist it'll create the plugin config with disablemediapreload only, and it's not right behaviour